hub/core/transform/transform.py

```diff
@@ -103,6 +107,13 @@ def store_shard(transform_input: Tuple):
     # separate cache for each tensor to prevent frequent flushing, 32 MB ensures only full chunks are written.
     storage_map = {key: LRUCache(MemoryProvider(), storage, 32 * MB) for key in tensors}
 
+    for tensor in tensors:
+        all_tensor_meta[tensor].htype = tensor_metas[tensor].htype
```

Comment:
I think we can probably merge this for loop with line 105 and initialize within the for loop here itself.